src/render.js

```diff
@@ -39,7 +39,8 @@ function render(
     const VueRouter = requiredRouter.default || requiredRouter
     localVue.use(VueRouter)
 
-    router = new VueRouter({routes})
+    router = new VueRouter({routes, mode: 'abstract'})
+    router.push('/')
```

Comment:
This felt like the most reasonable default to use, since in `abstract` mode the stack is empty. In the default `hash` mode with a clean JSDOM environment, this would also be the behavior.

Reply:
Hi! Let me take a closer look, as even though this fixes the issue, we might have some unexpected behavior here for users who would expect a regular, history-based router (not entirely sure what are the differences with `abstract` mode).

Maybe we could follow a similar approach to https://github.com/testing-library/vue-testing-library/pull/232, where we allow users to pass their own instance of Vuex. This would allow people to set their own router mode while also preserving backwards compatibility and consistency between vuex/router options. What do you think about it?

Reply:
It's a fair point that users might expect a hash-based / history router to be used with this library. In my case, at least, having to pass the entire Vue Router would be a bit of a pain to have to do with each test, as `abstract` mode is the configuration I'd prefer to use throughout all my tests. So, at least initially, my thought is that some kind of global config file would be useful, for example:

```js
// vue-testing-library.config.js
module.exports = {
  vueRouterMode: 'abstract',
};
```

However, it looks like other `testing-library` plugins don't have a global configuration file where a user could set something like this. For instance, `react-testing-library` has support for a custom `render` function ([docs](https://testing-library.com/docs/react-testing-library/setup/)) where the user can set defaults.

So perhaps the way forward is to allow passing an instantiated Vue Router, but then also provide docs for a custom `render` function like the `react-testing-library` docs above? Either way, I think it also makes sense to add a point to the "FAQs" page about this, as it's a bit confusing no matter what we decide to do.

I'm happy to tackle these updates depending on the maintainer's preferences. Thanks!

Reply:
> For instance, react-testing-library has support for a custom render function (docs) where the user can set defaults

We can do the same thing here at Vue Testing Lib:

```js
import VueRouter from 'vue-router'
import MyComponent from './MyComponent'
import routes from './routes'

function customRender() {
  const router = new VueRouter({ routes, mode: 'abstract' })
  router.push('/')

  return render(MyComponent, { routes: router })
}

// and then

test('foo', () => {
  const { getByText } = customRender()
})
```

This would allow people to set up their custom Router's mode a single time